server/weaverbird/pipeline_executor.py

```diff
@@ -22,13 +21,13 @@ class PipelineExecutor:
     def __init__(self, domain_retriever: DomainRetriever):
         self.retrieve_domain = domain_retriever
 
-    def execute_pipeline(self, pipeline_steps: List[dict]) -> DataFrame:
+    def execute_pipeline(self, pipeline: Pipeline) -> DataFrame:
```

Comment:
@fspot I think it's cleaner to pass a Pipeline and not a List[dict]. However, it leads to some circular imports. Is it worth it ?

Reply:
I think you can also use the string `'Pipeline'`. See [Forward references](https://www.python.org/dev/peps/pep-0484/#forward-references) section of the PEP 484 on type hints

Reply:
It seems quite logical than PipelineExecutor imports Pipeline.
Maybe the other way around isn't ^^ So you know what causes the import of PipelineExecutor by Pipeline? If it's easy, we can correct it.

Reply:
Pipeline imports steps
some steps (append, join mostly) use the pipeline executor
I think that we can workaround this by importing the PipelineExecutor for those step when they are called.

Reply:
Awww true
In that case, maybe we should import locally the pipeline executor in the execute method of these steps? That would break the circular import